pydm/widgets/image.py

```diff
@@ -159,14 +159,17 @@ def image_connection_state_changed(self, conn):
     def image_value_changed(self, new_image):
         """
         Callback invoked when the Image Channel value is changed.
-        Reshape and display the new image.
+        We try to do as little as possible in this method, because it
+        gets called every time the image channel updates, which might
+        be extremely often.  Basically just store the data, and set
+        a flag requesting that the image be redrawn.
 
         Parameters
         ----------
         new_image : np.ndarray
             The new image data.  This can be a flat 1D array, or a 2D array.
         """
-        if new_image is None:
+        if new_image is None or new_image.size == 0:
```

Comment:
Is `size` a scalar or a tuple? I guess scalar since it is the waveform, right?

Reply:
`size` is a scalar, regardless of array dimensions:
```
>>> b = np.array([[1,2,3],[4,5,6]])
>>> b.size
6
```

Reply:
Awesome! 👍